### DISCUSSION TITLE
how to create object in icub simulator using python language  ?

### DISCUSSION THREAD

**Discussion by hamaduddin** [robotology/community#591] [Q&A] [answered]
I am a masters student working on visual smooth pursuit of icub robot ,I am unable to create object by using python in icub simulator and want to create it and make it move in front of  icub eyes so that i can use it in my experiment by using python ,kindly help in this regards by some working example,Thanks .

**Comment by pattacini** [comment, marked as answer]
Hy @hamaduddin 

For our official simulator based on Gazebo, you could get an idea of how to spawn a sphere by inspecting this tutorial: https://github.com/vvv-school/tutorial_gaze-interface.

Some details:
- the model of the sphere is at https://github.com/vvv-school/tutorial_gaze-interface/tree/master/gazebo/models/tutorial_gaze-interface-ball.
- the code to make the sphere move is at https://github.com/vvv-school/tutorial_gaze-interface/blob/master/src/mover.cpp

Yes, it's a C++ codebase (and you should learn/use C++ for doing real robotics 😉), but I found convenient Python bindings to interact w/ Gazebo: check out https://pypi.org/project/pygazebo. You may peruse it.

---

Instead, if you're referring to our old `iCub_SIM`, here's the link to the doc:
- [http://wiki.icub.org/wiki/Simulator_README#OBJECT_INFORMATION:](http://wiki.icub.org/wiki/Simulator_README#OBJECT_INFORMATION:)

The instructions are based on the YARP plumbing, so following these [examples](https://github.com/robotology/yarp/tree/master/bindings/python/examples) should suffice.
